client/dashboard/sites/deployments-list/style.scss

```diff
@@ -0,0 +1,5 @@
+.dashboard-site-deployments-list {
+	.dataviews-view-table col.dataviews-view-table__col-commit {
+		width: auto;
```

Comment:
Spent way too much time on it. 😅 

Basically this is to revert the logic from https://github.com/WordPress/gutenberg/pull/72969/files#diff-90161488605f493d6d3373ccc278b42e484d27347f7e78794f182d181bf04050R319.

Otherwise, the Commit column will "lose" to the title field and will be shrunk as narrow as possible to make room for the title field like this. The branch name and badge will break:

<img width="1166" height="204" alt="Image" src="https://github.com/user-attachments/assets/26843495-ba83-4971-aa55-501edd09b4d6" />